ports/stm32/main.c

```diff
@@ -185,10 +185,17 @@ MP_NOINLINE static bool init_flash_fs(uint reset_mode) {
 
     if (len != -1) {
         // Detected a littlefs filesystem so create correct block device for it
-        mp_obj_t args[] = { MP_OBJ_NEW_QSTR(MP_QSTR_len), MP_OBJ_NEW_SMALL_INT(len) };
-        bdev = MP_OBJ_TYPE_GET_SLOT(&pyb_flash_type, make_new)(&pyb_flash_type, 0, 1, args);
+        mp_obj_t lfs_bdev = pyb_flash_new_obj(0, len);
+        if (lfs_bdev == mp_const_none) {
+            // invalid len detected, filesystem header block likely corrupted.
+            mp_printf(&mp_plat_print, "MPY: corrupted lfs filesystem\n");
+            // Create bdev with default size to allow user to attempt recovery if desired.
+            bdev = pyb_flash_new_obj(0, -1);
+            ret = true;
```

Comment:
This assignment to `ret` won't do anything, it's overwritten below.

Also, I would not print any error message, because:
1. if it actually succeeds to mount below because it really is a valid filesystem, then this message is confusing
2. it'll anyway print the "can't mount flash" error below if it really can't mount it after trying again

Reply:
Ah yes of course, I hadn't re-read the rest of the function. Cleaned up now.
Thanks for the test code too, I run that before and after this commit on pybd and reproduced the same results!

Reply:
Excellent, thanks!